### ISSUE TITLE
Add context to login hooks

### ISSUE THREAD

**Issue by rullzer** [nextcloud/server#9757]
Right now we have pre/post login hooks that are emitted.

However there is no context to those. It could be a login via the web. it could be a login via an apptoken etc.

Extending the current hooks. Or introducing a new event would allow apps to react to different kind of logins.

CC: @ChristophWurst

**Comment by ChristophWurst** [comment]
:+1: as discussed

Do we fire the post login hook right after the login or do we wait until the 2fa challenge has been passed?

**Comment by rullzer** [comment]
Actually at both cases some hook/event would make sense IMO.

**Comment by ChristophWurst** [comment]
Yes, because then you could actually detect an login attempt and the successful login (esp. in regards to #9750)

**Comment by szaimen** [comment]
I suppose this issue is still valid? If not, please close this issue!

**Comment by ChristophWurst** [comment]
This is still relevant: https://github.com/nextcloud/server/blob/master/lib/public/User/Events/BeforeUserLoggedInEvent.php
